### PR TITLE
Remove more dtos

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Remove more dtos that aren't needed anymore. Output dtos we can remove next in another PR.

## Links to any relevant issues

https://github.com/iotaledger/iota-sdk/issues/1713

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

cargo test --all-features

### CODE REVIEW

File: sdk/src/types/block/output/unlock_condition/storage_deposit_return.rs [hunk 1]
```diff
@@ -71,7 +71,7 @@ mod dto {
 
     #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
     #[serde(rename_all = "camelCase")]
-    pub struct StorageDepositReturnUnlockConditionDto {
+    pub struct StorageDepositReturnUnlockcondition {
```
Comment (thibault-martinez) [inline sdk/src/types/block/output/unlock_condition/storage_deposit_return.rs]: ?

Comment (Thoralf-M) [reply sdk/src/types/block/output/unlock_condition/storage_deposit_return.rs]: Changed too much

Comment (Thoralf-M) [reply sdk/src/types/block/output/unlock_condition/storage_deposit_return.rs]: Fixed

**Comment by DaughterOfMars** [comment]
conflicts

File: sdk/src/types/block/output/anchor.rs
Comment (DaughterOfMars) [inline sdk/src/types/block/output/anchor.rs:1]: can we make all the output dtos private?

Comment (Thoralf-M) [reply sdk/src/types/block/output/anchor.rs:1]: Yes, but this should be again in a new PR :trollface:

Comment (Thoralf-M) [reply sdk/src/types/block/output/anchor.rs:1]: Actually we can remove them I think

**Review by thibault-martinez — DISMISSED** [review]

File: sdk/src/types/block/output/anchor.rs
Comment (DaughterOfMars) [reply sdk/src/types/block/output/anchor.rs:1]: I'm confused, did you make this change and then revert it?

Comment (Thoralf-M) [reply sdk/src/types/block/output/anchor.rs:1]: I just renamed something in the output dto, but didn't actually remove anything there yet

Comment (DaughterOfMars) [reply sdk/src/types/block/output/anchor.rs:1]: well, they can still be made private, right?

Comment (Thoralf-M) [reply sdk/src/types/block/output/anchor.rs:1]: Why not remove them completely? https://github.com/iotaledger/iota-sdk/issues/1713#issuecomment-1834162589

Comment (DaughterOfMars) [reply sdk/src/types/block/output/anchor.rs:1]: They still need the type field

Comment (thibault-martinez) [reply sdk/src/types/block/output/anchor.rs:1]: I'm confused now, is there still something to be done here? Or another PR ?

Comment (DaughterOfMars) [reply sdk/src/types/block/output/anchor.rs:1]: I think we should go ahead and make these private here, as they don't need to be public afaict

Comment (thibault-martinez) [reply sdk/src/types/block/output/anchor.rs:1]: Maybe I'm wrong but I think the tests need them to be public?

Comment (thibault-martinez) [reply sdk/src/types/block/output/anchor.rs:1]: Can a tests mod be nested? We could move it inside the dto mod

Comment (DaughterOfMars) [reply sdk/src/types/block/output/anchor.rs:1]: Yes, they can

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]
beautiful stuff

**Pull request merged**